entity-framework/core/what-is-new/ef-core-3.0/breaking-changes.md

```diff
@@ -1711,3 +1712,38 @@ modelBuilder
      .HasOne(e => e.UpdatedBy)
      .WithMany();
 ```
+
+<a name="udf-empty-string"></a>
+### DbFunction.Schema being null or empty string configures it to be in model's default schema
+
+[Tracking Issue #12757](https://github.com/aspnet/EntityFrameworkCore/issues/12757)
+
+This change is introduced in EF Core 3.0-preview 7.
+
+**Old behavior**
+
+A DbFunction with value of schema being empty string configured the function to be in no schema and treated it as built-in function. For example following code will map `DatePart` CLR function to `DATEPART` built-in function on SqlServer.
+
+```C#
+[DbFunction("DATEPART", Schema = "")]
+public static int? DatePart(string datePartArg, DateTime? date) => throw new Exception();
+
+```
+
+**New behavior**
+
+All DbFunction mappings are considered to be mapped to user defined functions. Hence empty string value would put the function inside default schema for the model. It can be default schema configured explicitly via fluent API `modelBuilder.HasDefaultSchema()` or `dbo`.
+
+**Why**
+
+Earlier schema being empty was a way to treat that function is built-in but that logic is only applicable for SqlServer where built-in function does not belong to any schema.
```

Comment:
```suggestion
The logic that allowed the empty schema to indicate a built-in function wasn't consistent or discoverable and was only applicable to SQL Server.
 ```

Reply:
I am kind of finding this argument a bit hard to buy myself. Is the new behavior more useful and applicable across databases. What happens in PostgreSQL now?

Reply:
@roji to answer in detail.

It makes it more useful for other providers. PostgreSQL, even built-in functions have schema. And user defined functions can be used with schema if they are in default schema. Determination of built-in or not based on schema being empty string is arbitrary.